.github/workflows/deploy-examples.yml

```diff
@@ -0,0 +1,68 @@
+name: Deploy example apps to Fly
+
+on:
+  # Allow calling this workflow from other workflows
+  workflow_call:
+    inputs:
+      version:
+        type: string
+        description: "Wasp version to use for deployment (default to latest)"
+        required: false
+    secrets:
+      FLY_API_TOKEN:
+        description: "Fly API token for deploying apps"
+        required: true
+
+  # Allow calling this workflow manually from the GitHub UI
+  workflow_dispatch:
+    inputs:
+      version:
+        description: "Wasp version to use for deployment (default to latest)"
+        required: false
+
+  # TODO: REMOVE THIS BEFORE MERGING, JUST FOR TESTING
+  push:
+    branches:
+      - 2831-automatically-deploy-all-example-apps-on-release
+
+concurrency:
+  group: "deploy-examples"
+  cancel-in-progress: true
+
+jobs:
+  deploy:
+    strategy:
+      matrix:
+        projects:
+          - name: waspleau
+            dir: examples/waspleau
+            url: https://waspleau-app-client.fly.dev/
+
+          - name: websockets-realtime-voting
+            dir: examples/websockets-realtime-voting
+            url: https://websockets-voting-client.fly.dev/
```

Comment:
There's some duplication here - we have the same info in our project and the secrets. Any chance we can get rid of it?

Reply:
Can you be more specific?

Reply:
Sure, here goes.

Keep in mind that if something sounds like it doesn't make sense, it's likely my lack of knowledge :)

What looks like duplication:
1. We list our example app here, and we list them in the `examples` folder.
2. The app URL is explicitly listed here, but it's also implicitly defined in our `fly-client.toml` and `fly-server.toml`

Is there a way we can avoid changing this action when we change the `fly-*.toml` files or the example's name.

Ideally, this is what I'd like (not sure if possible):
1. Each app has its deployment workflow that triggers when that app is, and is defined in the respective app's folder.
2. There's no duplication of app names (e.g., `waspleau`) or Fly URLs (https://websockets-voting-client.fly.dev/).

Reply:
> Each app has its deployment workflow that triggers when that app is, and is defined in the respective app's folder (I'm guessing this would make it harder to trigger all of them, but it would also give us the option to trigger only one).

Mmmm having the deployment workflow in each folder is not possible, they all need to be in `<root>/.github/workflows`. We could create a new workflow per app, or require all of them have a `deploy.sh` script, but tbh it does look like overkill. We just need to run `wasp deploy fly deploy` in the folders.

> There's no duplication of app names (e.g., waspleau) or Fly URLs 

Done